pydicom/dataset.py

```diff
@@ -342,6 +342,12 @@ def __init__(self, *args, **kwargs):
         # known private creator blocks
         self._private_blocks = {}
 
+        for tag, data_elem in self._dict.items():
+            if not isinstance(data_elem, DataElement):
+                # Don't load the element if it has been deferred
+                if data_elem.value is not None:
+                    self._initial_dataelement_handling(tag, data_elem)
```

Comment:
This is what I originally tried. It fixed the issue for my subset of cases, but it caused a large number of the tests to fail. So not sure how to go about this.